controller/external-secrets/application.yaml

```diff
@@ -0,0 +1,26 @@
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: external-secrets
+  namespace: kubezero
+  annotations:
+    argocd.argoproj.io/sync-wave: '50'
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io
```

Comment:
do we need to specify this? isn't argo already auto inject it?

Reply:
I'm explicitly setting the Argo CD app [deletion finalizer](https://argo-cd.readthedocs.io/en/latest/user-guide/app_deletion/#about-the-deletion-finalizer).

By default, it's `foreground cascading deletion`, so I'm kind of stating the default here, but making it more visible.